transform-plugins/src/main/java/co/cask/hydrator/plugin/XMLParser.java

```diff
@@ -0,0 +1,314 @@
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathFactory;
+
+/**
+ * Parses XML Event using XPath.
+ * This should generally be used in conjunction with the XML Reader Batch Source.
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name("XMLParser")
+@Description("Parse XML events based on XPath")
+public class XMLParser extends Transform<StructuredRecord, StructuredRecord> {
+  private final Config config;
+  private Schema outSchema;
+  private Map<String, String> xPathMapping = new HashMap<>();
+  private Map<String, Integer> mapErrorProcessing = new HashMap<>();
+
+  // Required only for testing.
+  public XMLParser(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    super.configurePipeline(pipelineConfigurer);
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getOutputSchema());
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    outSchema = config.getOutputSchema();
+    validateXpathAndSchema();
+    mapErrorProcessing.put("Ignore error and continue", 1);
+    mapErrorProcessing.put("Exit on error", 2);
+    mapErrorProcessing.put("Write to error dataset", 3);
+  }
+
+  /**
+   * Valid if xpathMappings and schema contain the same field names.
+   */
+  private void validateXpathAndSchema() {
+    String[] xpaths = config.xpathFieldMapping.split(",");
+    String fieldName;
+    for (String xpath : xpaths) {
+      String[] xpathmap = xpath.split(":"); //name:xpath[,name:xpath]*
+      fieldName = xpathmap[0].trim();
+      if (Strings.isNullOrEmpty(fieldName)) {
+        throw new IllegalArgumentException("Field name cannot be null or empty.");
+      } else if (xpathmap.length < 2 || Strings.isNullOrEmpty(xpathmap[1])) {
+        throw new IllegalArgumentException(String.format("Xpath for field name, %s cannot be null or empty.",
+                                                         fieldName));
+      }
+      xPathMapping.put(fieldName, xpathmap[1].trim());
+    }
+    List<Schema.Field> outFields = outSchema.getFields();
+    // Checks if all the fields in the XPath mapping are present in the output schema.
+    // If they are not a list of fields that are not present is included in the error message.
+    StringBuilder notOutput = new StringBuilder();
+    for (Schema.Field field : outFields) {
+      fieldName = field.getName();
+      if (!xPathMapping.keySet().contains(field.getName())) {
+        notOutput.append(fieldName + ";");
+      }
+    }
+    if (notOutput.length() > 0) {
+      throw new IllegalArgumentException("Following fields are not present in output schema :" +
+                                           notOutput.toString());
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) {
+    String fieldName;
```

Comment:
I don't think it helps to keep these variable declarations separate from the assignments.